RenderLayerScrollableArea::updateResizerStyle should early return if there is no resizer

Ahmad-S792 opened the pull request:
#### b5f2eee537a0cfdb71b7df4a2070a665b0c24e2b
<pre>
RenderLayerScrollableArea::updateResizerStyle should early return if there is no resizer

RenderLayerScrollableArea::updateResizerStyle should early return if there is no resizer
<a href="https://bugs.webkit.org/show_bug.cgi?id=251676">https://bugs.webkit.org/show_bug.cgi?id=251676</a>

Reviewed by Simon Fraser.

Merge - <a href="https://chromium.googlesource.com/chromium/blink/+/1add5ffe5081cf282014bc4cfe04b8637244c607">https://chromium.googlesource.com/chromium/blink/+/1add5ffe5081cf282014bc4cfe04b8637244c607</a>

This is potential optimization by adding early return to &quot;updateResizeStyle&quot; function in case if there is no resizer.

* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(RenderLayerScrollableArea::updateResizerStyle): Add early return

Canonical link: <a href="https://commits.webkit.org/260857@main">https://commits.webkit.org/260857@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4d3168a1a3473d1e4210f5ce36389bd99ff3b2b1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107243 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16287 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40256 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116401 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115862 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111143 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17736 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7557 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99401 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113008 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13385 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96630 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40997 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95286 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28257 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/82768 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9348 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29609 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9985 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6453 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15503 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49190 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7528 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11518 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mdubet commented on Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
```diff
@@ -1751,6 +1752,9 @@ void RenderLayerScrollableArea::clearScrollCorner()
 
 void RenderLayerScrollableArea::updateResizerStyle()
 {
+    if (!m_resizer && !m_layer.canResize())
```
Do we really need both conditions to early return or any of them (`&&` or `||` ) ?

Ahmad-S792 replied:
Based on my understanding of code, we need both since `m_resize` is for custom scrollbar resized while `canResize()` is for special case we have for 'iframe'. As for `&&` or `||`, I think `||` will cover more cases but I wanted to be failsafe and only kept similar to what Blink commit did. Happy to change and push new commit. I am doing local build with `||` in background to confirm it works and does not cause any issue.

Ahmad-S792 replied:
Local build across various website seems to work fine and does not seems have any broken resizing behavior while changing to `||` and it would cover both cases while if either of them are true, which makes more sense. So I pushed new commit with change. Let's see what happens in EWS. Thanks for review!

Ahmad-S792 replied:
Since `||` will cover more cases in individually. I would opt for it and it does not regress anything on EWS as well. Marking this as "RESOLVED".

smfr commented on Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
```diff
@@ -1751,6 +1752,9 @@ void RenderLayerScrollableArea::clearScrollCorner()
 
 void RenderLayerScrollableArea::updateResizerStyle()
 {
+    if (!m_resizer || !m_layer.canResize())
```
This seems wrong. The function can create an `m_resizer` below, and maybe `layer.canResize()` changed since last time, requiring us to remove the existing resizer.

Ahmad-S792 replied:
Hmm. I think "&&" make sense. Let me push the original update. Unless you think this early return in the concept is wrong then I can close this PR.

Ahmad-S792 replied:
@smfr - `mac-wk1` seems to have issue so this failure is not related to this patch.

Ahmad-S792 replied:
@smfr - Any update? Or can I close this PR?

Ahmad-S792 committed:
RenderLayerScrollableArea::updateResizerStyle should early return if there is no resizer

RenderLayerScrollableArea::updateResizerStyle should early return if there is no resizer
https://bugs.webkit.org/show_bug.cgi?id=251676

Reviewed by Simon Fraser.

Merge - https://chromium.googlesource.com/chromium/blink/+/1add5ffe5081cf282014bc4cfe04b8637244c607

This is potential optimization by adding early return to "updateResizeStyle" function in case if there is no resizer.

* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(RenderLayerScrollableArea::updateResizerStyle): Add early return

Canonical link: https://commits.webkit.org/260857@main
